samtranslator/utils/utils.py

```diff
@@ -49,3 +49,21 @@ def dict_deep_get(d: Any, path: str) -> Optional[Any]:
         relative_path = (relative_path + f".{_path_nodes[0]}").lstrip(".")
         _path_nodes = _path_nodes[1:]
     return d
+
+
+def dict_deep_update(d: Any, path: str, dict_to_merge: Dict[str, Any]) -> None:
+    """
+    Update the value deep in the dict.
+
+    If any value along the path doesn't exist, create empty dict along the way.
```

Comment:
I'd find it more intuitive if `dict_to_merge` included the "path", so it's always just a merge of two dicts. The `path` just adds confusion IMO.

Reply:
interesting

Reply:
Good idea, adopted

Reply:
The new version includes the literal `path` in the key?

I meant instead of:

```
merge(d, "foo.bar", e)
```

Do:

```
merge(d, {"foo": {"bar": e}})
```

Reply:
this works better

Reply:
wait, what you proposed is not flexible. It doesn't allow me to control down to which level I want to do a "replace" instead of "merge".

For example, let's say, I want to set the `a.b.c` to `{"Mode": "Flexible"}. If the original dict already has `a.b.c` set to `{"Disabled": True}`, that old key/value will not be removed

Reply:
I am getting rid of this deep update thingy. It is not worth it

Reply:
resolving this one as we don't `dict_deep_update` anymore